### PR TITLE
docs: sphinx: Add units documentation page

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
This is primarily a simple overview, with links to @dgault's examples.

### CODE REVIEW

File: docs/sphinx/developers/units.txt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+solution to this was to add a unit of measurement to each of these
+properties.  The image size, for example, was previously specified to
+be stored in micrometers but may now be specified in any SI length
+unit of choice, or one of the supported non-SI length units.  This
+permits the preservation of the unit used by a proprietary file format
+or used at acquisition time, for example nanometers, millimeters,
+meters, or inches or thousandths of an inch could be used instead.
+
+At the OME-XML level, the properties continue to use the old names.
+For example, the ``PhysicalSizeX`` attribute.  They are supplemented
```
Comment (mtbc) [inline docs/sphinx/developers/units.txt]: not a sentence

**Comment by rleigh-codelibre** [comment]
@mtbc Should be fixed now.

File: docs/sphinx/developers/units.txt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+Since Bio-Formats 5.1 and the adoption of the 2015-01 OME Data Model,
+the data model and the corresponding Bio-Formats model and metadata
+APIs have added support for units of measurement.  Previously, the
+units for various properties such as the physical size of an image,
+stage position, confocal pinhole size, light wavelengths etc. were
+fixed in the model.  This was however somewhat inflexible, and not
+appropriate for imaging modalities at widely different scales, and the
+solution to this was to add a unit of measurement to each of these
+properties.  The image size, for example, was previously specified to
```
Comment (dgault) [inline docs/sphinx/developers/units.txt]: I would probably break this into 2 sentences :
This was however somewhat inflexible, and not appropriate for imaging modalities at widely different scales. The solution to this was to add a unit of measurement to each of these properties.

**Comment by dgault** [comment]
Other than that one comment to the new units page looks and displays fine and the link from the index is working. 

The content covers everything I can think of, reads well and gives a good overview for users. The links to the various examples all work correctly also.

**Comment by rleigh-codelibre** [comment]
Fixed, thanks.

**Comment by dgault** [comment]
With the fixes Im happy with the new units documentation in this PR. Good to merge once Travis is green.

**Pull request merged**